File: Backend/src/main/java/com/AgendaMed/Backend/model/Consulta.java
package com.AgendaMed.Backend.model;

import java.time.LocalDateTime;

import com.AgendaMed.Backend.model.enums.StatusConsulta;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "consulta")
public class Consulta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "medico_fk", nullable = false)
    private Medico medico;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "paciente_fk", nullable = false)
    private Paciente paciente;

    @Future
    @NotNull
    private LocalDateTime dataHora;

    @NotNull
    @Enumerated(EnumType.STRING)
    private StatusConsulta status = StatusConsulta.AGENDADA;
}

File: frontend/types/consulta.ts
import { MedicoResumoDTO } from './medico';
import { PacienteResumoDTO } from './paciente';

export type StatusConsulta = 'AGENDADA' | 'CANCELADA' | 'REALIZADA';

export interface ConsultaResponseDTO {
    id: number;
    medico: MedicoResumoDTO;
    paciente: PacienteResumoDTO;
    dataHora: string;
    status: StatusConsulta;
}

export interface AgendaDTO {
    consulta: ConsultaResponseDTO;
    proximasConsultas: ConsultaResponseDTO[];
}